### PR TITLE
chore(tracing): adds `out.host` tag for celery

### PR DESCRIPTION
**Pull request by zarirhamza**
Adds `out.host` tag for celery as a precursor tag for `peer.service`
Testing is updated to check for new tag by calling hostname for local machine

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/contrib/celery/signals.py [hunk 1]
```diff
@@ -156,6 +160,11 @@ def trace_after_publish(*args, **kwargs):
     if span is None:
         return
     else:
+        nodename = span.get_tag("celery.hostname")
+        if nodename is not None:
+            _, host = nodenames.nodesplit(nodename)
+            span.set_tag_str(net.TARGET_HOST, host)
```
Comment (jbertran) [inline ddtrace/contrib/celery/signals.py:166]: I gave you the typo'd version of my diff 🤦 
```suggestion
            span.set_tag_str(net.TARGET_HOST, nodename)
```

File: ddtrace/contrib/celery/signals.py [hunk 2]
```diff
@@ -124,6 +126,8 @@ def trace_before_publish(*args, **kwargs):
     span.set_tag_str(c.TASK_TAG_KEY, c.TASK_APPLY_ASYNC)
     span.set_tag_str("celery.id", task_id)
     set_tags_from_context(span, kwargs)
+    if kwargs.get("headers") is not None:
+        set_tags_from_context(span, kwargs["headers"])
```
Comment (mabdinur) [inline ddtrace/contrib/celery/signals.py:130]:
```suggestion
    if kwargs.get("headers") is not None:
        set_tags_from_context(span, kwargs["headers"])
```

Comment (mabdinur) [inline ddtrace/contrib/celery/signals.py:130]:
```suggestion
    if kwargs.get("headers") is not None:
        set_tags_from_context(span, kwargs["headers"])
```

**Review by mabdinur — APPROVED** [review]

**Review by mabdinur — CHANGES_REQUESTED** [review]
We should off on merging this PR until the Github incident is resolved.

File: releasenotes/notes/celery-precursor-peer.service-d3b1bc05f1ca59ab.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    celery: Resolves an issue where hostname tags were not set in spans generated by ``celery>4.0``.
+
```
Comment (mabdinur) [inline releasenotes/notes/celery-precursor-peer.service-d3b1bc05f1ca59ab.yaml:5]:
```suggestion
```

**Review by mabdinur — APPROVED** [review]

**Pull request merged**